core/trino-main/src/test/java/io/trino/server/ui/TestWebUi.java

```diff
@@ -677,8 +682,32 @@ public void testOAuth2AuthenticatorWithoutOpenIdScope()
                         .setBinding()
                         .toInstance(oauthClient))
                 .build()) {
-            HttpServerInfo httpServerInfo = server.getInstance(Key.get(HttpServerInfo.class));
-            assertAuth2Authentication(httpServerInfo, oauthClient.getAccessToken());
+            assertAuth2Authentication(server, oauthClient.getAccessToken(), false);
+        }
+        finally {
+            jwkServer.stop();
+        }
+    }
+
+    @Test
+    public void testOAuth2AuthenticatorWithRefreshToken()
+            throws Exception
+    {
+        OAuth2ClientStub oauthClient = new OAuth2ClientStub(false);
+        TestingHttpServer jwkServer = createTestingJwkServer();
+        jwkServer.start();
+        try (TestingTrinoServer server = TestingTrinoServer.builder()
+                .setProperties(ImmutableMap.<String, String>builder()
+                        .putAll(OAUTH2_PROPERTIES)
+                        .put("http-server.authentication.oauth2.jwks-url", jwkServer.getBaseUrl().toString())
+                        .put("http-server.authentication.oauth2.refresh-tokens", "true")
+                        .put("http-server.authentication.oauth2.refresh-tokens.issued-token.timeout", REFRESH_TOKEN_TIMEOUT_MINUTES + "m")
```

Comment:
This is a timeout for the token minted by us (i.e with access token and refresh token) - can we have a test where the minted refresh token is expired ?

Reply:
I added additional test for that: `testOAuth2AuthenticatorRefreshTokenExpiration`